models/decorators_roles.py

```diff
@@ -0,0 +1,48 @@
+from flask import session, redirect, url_for
+from functools import wraps
+from models.users import User
+
+
+def require_logged_in(redirect_page='library.login'):
+    def decorator(func):
+        @wraps(func)
+        def inner_func(*args, **kwargs):
+            if 'logged_in' in session:
+                if session['logged_in']:
```

Comment:
I guess the two `if`s could be reduced to one `if`.

Reply:
Roger that